packages/@mantine/core/src/components/TagsInput/TagsInput.tsx

```diff
@@ -336,7 +336,7 @@ export const TagsInput = factory<TagsInputFactory>((_props, ref) => {
       key={`${item}-${index}`}
       withRemoveButton={!readOnly}
       onRemove={() => {
-        setValue(_value.filter((i) => item !== i));
+        setValue(_value.splice(index, 1));
```

Comment:
Is this correct? I'm seeing that clicking X to remove a tag now removes all other tags except the target

Reply:
think I should've used `.toSpliced` here @adelamarAtGalileo can you let me know if this resolves it? 

Seems alright to me, but clearly I missed it yesterday. Storybook must not have refreshed :shrug:.

Reply:
`splice` modifies the array and returns the removed value

![Screenshot 2024-09-30 at 1 55 04 PM](https://github.com/user-attachments/assets/14a869ba-1eb9-4596-8e29-d84064b10a2e)

`toSpliced` doesn't modify the array and returns the remaining values

![Screenshot 2024-09-30 at 1 57 00 PM](https://github.com/user-attachments/assets/bceeb489-16cb-405e-9786-f4bf7868a5b9)

So yes, `.toSpliced` should work here

Reply:
nice, hopefully released soon, already got complaints from users :D

Reply:
@max-degterev You can downgrade to  `"@mantine/core": "^7.11.1"` you can avoid the newly introduced bug while waiting for a new release with the fix

Reply:
maybe @rtivital pushes this out soon, I keep my hopes up :)